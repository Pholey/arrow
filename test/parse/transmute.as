
// Transmute
let a = 0 as *int8;
let b = 20;
let c = &x as *int128;
let d = c as int64;
