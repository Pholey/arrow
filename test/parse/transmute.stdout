{
  "tag": "Module",
  "span": "test/parse/transmute.as:1-6",
  "statements": [
    {
      "tag": "Variable",
      "span": "test/parse/transmute.as:3:1-20",
      "name": "a",
      "mutable": false,
      "type": null,
      "initializer": {
        "tag": "Transmute",
        "span": "test/parse/transmute.as:3:11-19",
        "operand": {
          "tag": "Integer",
          "span": "test/parse/transmute.as:3:9-10",
          "value": "0"
        },
        "target": {
          "tag": "TypePointer",
          "span": "test/parse/transmute.as:3:14-19",
          "element": {
            "tag": "TypeName",
            "span": "test/parse/transmute.as:3:15-19",
            "name": {
              "tag": "Identifier",
              "span": "test/parse/transmute.as:3:15-19",
              "text": "int8"
            }
          }
        }
      }
    },
    {
      "tag": "Variable",
      "span": "test/parse/transmute.as:4:1-12",
      "name": "b",
      "mutable": false,
      "type": null,
      "initializer": {
        "tag": "Integer",
        "span": "test/parse/transmute.as:4:9-11",
        "value": "20"
      }
    },
    {
      "tag": "Variable",
      "span": "test/parse/transmute.as:5:1-23",
      "name": "c",
      "mutable": false,
      "type": null,
      "initializer": {
        "tag": "Transmute",
        "span": "test/parse/transmute.as:5:12-22",
        "operand": {
          "tag": "AddressOf",
          "span": "test/parse/transmute.as:5:9-11",
          "operand": {
            "tag": "Identifier",
            "span": "test/parse/transmute.as:5:10-11",
            "text": "x"
          }
        },
        "target": {
          "tag": "TypePointer",
          "span": "test/parse/transmute.as:5:15-22",
          "element": {
            "tag": "TypeName",
            "span": "test/parse/transmute.as:5:16-22",
            "name": {
              "tag": "Identifier",
              "span": "test/parse/transmute.as:5:16-22",
              "text": "int128"
            }
          }
        }
      }
    },
    {
      "tag": "Variable",
      "span": "test/parse/transmute.as:6:1-20",
      "name": "d",
      "mutable": false,
      "type": null,
      "initializer": {
        "tag": "Transmute",
        "span": "test/parse/transmute.as:6:11-19",
        "operand": {
          "tag": "Identifier",
          "span": "test/parse/transmute.as:6:9-10",
          "text": "c"
        },
        "target": {
          "tag": "TypeName",
          "span": "test/parse/transmute.as:6:14-19",
          "name": {
            "tag": "Identifier",
            "span": "test/parse/transmute.as:6:14-19",
            "text": "int64"
          }
        }
      }
    }
  ]
}
